From 932cf36576de29e74b3aafb90ba20f92397344af Mon Sep 17 00:00:00 2001
From: wengzm0704 <wengzm0704@thundersoft.com>
Date: Fri, 14 Jul 2017 10:56:15 +0800
Subject: [PATCH] Support LCD BOE BV055HDM-L01-1S01 FT8006M on QRD8953 driver

BASE_BUILD_ID: LA.UM.5.6.r1-04700-89xx.0-1
BASE_PLATFORM: QRD8953_RD10
BASE_GIT_PROJECT: kernel/msm-3.18
BASE_GIT_BRANCH: android-external.lnx.2.0.r5-rel
BASE_BRANCH_DIR: kernel
BASE_AU: AU_LINUX_ANDROID_LA.UM.5.5.R1.07.01.02.269.047

Change-Id: I6e85187de19fe01b7b80dcf3ab27b87d285f04ba
---
 .../dts/qcom/dsi-panel-ft8006m-boe-720p-video.dtsi |  77 ++++++++++++
 arch/arm/boot/dts/qcom/msm8953-mdss-panels.dtsi    |   9 ++
 arch/arm/boot/dts/qcom/msm8953-qrd-sku3.dtsi       |   6 +-
 .../arm/boot/dts/qcom/panel_ft8006m_720p_video.xml | 138 +++++++++++++++++++++
 4 files changed, 229 insertions(+), 1 deletion(-)
 create mode 100755 arch/arm/boot/dts/qcom/dsi-panel-ft8006m-boe-720p-video.dtsi
 mode change 100644 => 100755 arch/arm/boot/dts/qcom/msm8953-mdss-panels.dtsi
 mode change 100644 => 100755 arch/arm/boot/dts/qcom/msm8953-qrd-sku3.dtsi
 create mode 100755 arch/arm/boot/dts/qcom/panel_ft8006m_720p_video.xml

diff --git a/arch/arm/boot/dts/qcom/dsi-panel-ft8006m-boe-720p-video.dtsi b/arch/arm/boot/dts/qcom/dsi-panel-ft8006m-boe-720p-video.dtsi
new file mode 100755
index 0000000..4ba2f3b
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/dsi-panel-ft8006m-boe-720p-video.dtsi
@@ -0,0 +1,77 @@
+/* Copyright (c) 2013, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/*---------------------------------------------------------------------------
+ * This file is autogenerated file using gcdb parser. Please do not edit it.
+ * Update input XML file to add a new entry or update variable in this file
+ * VERSION = "1.0"
+ *---------------------------------------------------------------------------*/
+&mdss_mdp {
+	dsi_ft8006m_boe_720p_video: qcom,mdss_dsi_ft8006m_boe_720p_video {
+		qcom,mdss-dsi-panel-name = "ft8006m boe vivo vdo mode dsi panel";
+		qcom,mdss-dsi-panel-controller = <&mdss_dsi0>;
+		qcom,mdss-dsi-panel-type = "dsi_video_mode";
+		qcom,mdss-dsi-panel-destination = "display_1";
+		qcom,mdss-dsi-panel-framerate = <60>;
+		qcom,mdss-dsi-virtual-channel-id = <0>;
+		qcom,mdss-dsi-stream = <0>;
+		qcom,mdss-dsi-panel-width = <720>;
+		qcom,mdss-dsi-panel-height = <1280>;
+		qcom,mdss-dsi-h-front-porch = <48>;
+		qcom,mdss-dsi-h-back-porch = <48>;
+		qcom,mdss-dsi-h-pulse-width = <16>;
+		qcom,mdss-dsi-h-sync-skew = <0>;
+		qcom,mdss-dsi-v-back-porch = <32>;
+		qcom,mdss-dsi-v-front-porch = <75>;
+		qcom,mdss-dsi-v-pulse-width = <8>;
+		qcom,mdss-dsi-h-left-border = <0>;
+		qcom,mdss-dsi-h-right-border = <0>;
+		qcom,mdss-dsi-v-top-border = <0>;
+		qcom,mdss-dsi-v-bottom-border = <0>;
+		qcom,mdss-dsi-bpp = <24>;
+		qcom,mdss-dsi-color-order = "rgb_swap_rgb";
+		qcom,mdss-dsi-underflow-color = <0xff>;
+		qcom,mdss-dsi-border-color = <0>;
+		qcom,mdss-dsi-on-command = [05 01 00 00 00 00 02 35 00		
+			29 01 00 00 00 00 02 09 00	
+			05 01 00 00 C8 00 02 11 00
+			05 01 00 00 78 00 02 29 00];
+		qcom,mdss-dsi-off-command = [05 01 00 00 32 00 02 28 00
+					05 01 00 00 78 00 02 10 00];
+		qcom,mdss-dsi-on-command-state = "dsi_lp_mode";
+		qcom,mdss-dsi-off-command-state = "dsi_lp_mode";
+		qcom,mdss-dsi-h-sync-pulse = <1>;
+		qcom,mdss-dsi-traffic-mode = "non_burst_sync_event";//burst_mode 2//non_burst_sync_event 1//non_burst_sync_pulse
+		qcom,mdss-dsi-lane-map = "lane_map_0123";
+		qcom,mdss-dsi-bllp-eof-power-mode;
+		qcom,mdss-dsi-bllp-power-mode;
+		qcom,mdss-dsi-lane-0-state;
+		qcom,mdss-dsi-lane-1-state;
+		qcom,mdss-dsi-lane-2-state;
+		qcom,mdss-dsi-lane-3-state;
+		qcom,mdss-dsi-panel-timings-phy-v2 = [1e 1b 04 06 02 03 04 a0
+					1e 1b 04 06 02 03 04 a0
+					1e 1b 04 06 02 03 04 a0
+					1e 1b 04 06 02 03 04 a0
+					1e 0e 03 05 02 03 04 a0];
+		qcom,mdss-dsi-t-clk-post = <0x0a>;
+		qcom,mdss-dsi-t-clk-pre = <0x1d>;
+		qcom,mdss-dsi-bl-min-level = <1>;
+		qcom,mdss-dsi-bl-max-level = <4095>;
+		qcom,mdss-dsi-dma-trigger = "trigger_sw";
+		qcom,mdss-dsi-mdp-trigger = "none";
+		qcom,mdss-dsi-bl-pmic-control-type = "bl_ctrl_wled";
+		qcom,mdss-dsi-reset-sequence = <1 20>, <0 20>, <1 20>;
+		qcom,mdss-dsi-lp11-init;
+		qcom,suspend-ulps-enabled;
+		};
+};
diff --git a/arch/arm/boot/dts/qcom/msm8953-mdss-panels.dtsi b/arch/arm/boot/dts/qcom/msm8953-mdss-panels.dtsi
old mode 100644
new mode 100755
index 14d742d..007b75d
--- a/arch/arm/boot/dts/qcom/msm8953-mdss-panels.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8953-mdss-panels.dtsi
@@ -21,6 +21,7 @@
 #include "dsi-panel-r69006-1080p-video.dtsi"
 #include "dsi-panel-r69006-1080p-cmd.dtsi"
 #include "dsi-panel-truly-wuxga-video.dtsi"
+#include "dsi-panel-ft8006m-boe-720p-video.dtsi"
 
 &soc {
 	dsi_panel_pwr_supply: dsi_panel_pwr_supply {
@@ -82,6 +83,14 @@
 		23 1a 08 09 05 03 04 a0];
 };
 
+&dsi_ft8006m_boe_720p_video {
+	qcom,mdss-dsi-panel-timings-phy-v2 = [1e 1b 04 06 02 03 04 a0
+		1e 1b 04 06 02 03 04 a0
+		1e 1b 04 06 02 03 04 a0
+		1e 1b 04 06 02 03 04 a0
+		1e 0e 03 05 02 03 04 a0];
+};
+
 &dsi_r69006_1080p_video {
 	qcom,mdss-dsi-panel-timings-phy-v2 = [24 1f 08 09 05 03 04 a0
 		24 1f 08 09 05 03 04 a0
diff --git a/arch/arm/boot/dts/qcom/msm8953-qrd-sku3.dtsi b/arch/arm/boot/dts/qcom/msm8953-qrd-sku3.dtsi
old mode 100644
new mode 100755
index 871e7f2..0bb6b44
--- a/arch/arm/boot/dts/qcom/msm8953-qrd-sku3.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8953-qrd-sku3.dtsi
@@ -71,7 +71,7 @@
 	ibb-supply = <&ibb_regulator>;
 	/delete-property/ vdd-supply;
 
-	qcom,dsi-pref-prim-pan = <&dsi_r69006_1080p_cmd>;
+	qcom,dsi-pref-prim-pan = <&dsi_ft8006m_boe_720p_video>; /* <&dsi_r69006_1080p_cmd>; */
 	pinctrl-names = "mdss_default", "mdss_sleep";
 	pinctrl-0 = <&mdss_dsi_active &mdss_te_active>;
 	pinctrl-1 = <&mdss_dsi_suspend &mdss_te_suspend>;
@@ -96,6 +96,10 @@
 	qcom,panel-supply-entries = <&dsi_panel_pwr_supply>;
 };
 
+&dsi_ft8006m_boe_720p_video {
+	qcom,panel-supply-entries = <&dsi_panel_pwr_supply>;
+};
+
 &int_codec {
 	status = "disabled";
 };
diff --git a/arch/arm/boot/dts/qcom/panel_ft8006m_720p_video.xml b/arch/arm/boot/dts/qcom/panel_ft8006m_720p_video.xml
new file mode 100755
index 0000000..821fd67
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/panel_ft8006m_720p_video.xml
@@ -0,0 +1,138 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+# Copyright (c) 2013, The Linux Foundation. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+#       copyright notice, this list of conditions and the following
+#       disclaimer in the documentation and/or other materials provided
+#       with the distribution.
+#     * Neither the name of The Linux Foundation nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED
+# WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+-->
+
+<GCDB>
+  <Version>"1.0"</Version>
+  <PanelId>ft8006m-boe-720p-video</PanelId>
+  <PanelH>ft8006m_boe_720p_video</PanelH>
+  <PanelEntry>
+
+	<!-- Panel configuration -->
+	<PanelName>"ft8006m boe vivo vdo mode dsi panel"</PanelName>
+	<PanelController>"mdss_dsi0"</PanelController>
+	<PanelInterface>10</PanelInterface>
+	<PanelType>0</PanelType>
+	<PanelDestination>"DISPLAY_1"</PanelDestination>
+	<PanelOrientation>0</PanelOrientation>
+	<PanelFrameRate>60</PanelFrameRate>
+	<PanelChannelId>0</PanelChannelId>
+	<DSIVirtualChannelId>0</DSIVirtualChannelId>
+	<PanelBroadcastMode>0</PanelBroadcastMode>
+	<!-- Optional Panel configuration -->
+	<!--BitClockFrequency>0</BitClockFrequency -->
+	<DSIStream>0</DSIStream>
+	<PanelCompatible>"qcom,mdss-dsi-panel"</PanelCompatible>
+	<InterleaveMode>0</InterleaveMode>
+
+	<!-- Panel Resolution -->
+	<PanelWidth>720</PanelWidth>
+	<PanelHeight>1280</PanelHeight>
+	<HFrontPorch>48</HFrontPorch>
+	<HBackPorch>48</HBackPorch>
+	<HPulseWidth>16</HPulseWidth>
+	<HSyncSkew>0</HSyncSkew>
+	<VBackPorch>32</VBackPorch>
+	<VFrontPorch>75</VFrontPorch>
+	<VPulseWidth>8</VPulseWidth>
+	<HLeftBorder>0</HLeftBorder>
+	<HRightBorder>0</HRightBorder>
+	<VTopBorder>0</VTopBorder>
+	<VBottomBorder>0</VBottomBorder>
+	<!-- Optional Panel resolution configuration -->
+	<!--HActiveRes>0</HActiveRes>
+	<VActiveRes>100</VActiveRes>
+	<InvertDataPolarity>0</InvertDataPolarity>
+	<InvertVsyncPolarity>0</InvertVsyncPolarity>
+	<InvertHsyncPolarity>0</InvertHsyncPolarity -->
+
+	<!-- Panel Color Information -->
+	<ColorFormat>24</ColorFormat>
+	<ColorOrder>0</ColorOrder>
+	<UnderFlowColor>0xff</UnderFlowColor>
+	<BorderColor>0</BorderColor>
+
+	<!-- Panel Command information -->
+	<OnCommand>"0x05, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x35, 0x00,
+			0x29, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x09, 0x00,	
+			0x05, 0x01, 0x00, 0x00, 0xC8, 0x00, 0x02, 0x11, 0x00,
+			0x05, 0x01, 0x00, 0x00, 0x78, 0x00, 0x02, 0x29, 0x00"</OnCommand>
+	<OffCommand>"0x05, 0x01, 0x00, 0x00, 0x32, 0x00, 0x02, 0x28, 0x00,
+			0x05, 0x01, 0x00, 0x00, 0x78, 0x00, 0x02, 0x10, 0x00"</OffCommand>
+	<OnCommandState>0</OnCommandState>
+	<OffCommandState>0</OffCommandState>
+
+	<!-- Video mode panel information -->
+	<HSyncPulse>1</HSyncPulse>
+	<HFPPowerMode>0</HFPPowerMode>
+	<HBPPowerMode>0</HBPPowerMode>
+	<HSAPowerMode>0</HSAPowerMode>
+	<BLLPEOFPowerMode>1</BLLPEOFPowerMode>
+	<BLLPPowerMode>1</BLLPPowerMode>
+	<TrafficMode>1</TrafficMode>
+	<DMADelayAfterVsync>0</DMADelayAfterVsync>
+	<BLLPEOFPower>0x9</BLLPEOFPower>
+
+	<!-- Lane Configuration -->
+	<DSILanes>4</DSILanes>
+	<DSILaneMap>0</DSILaneMap>
+	<Lane0State>1</Lane0State>
+	<Lane1State>1</Lane1State>
+	<Lane2State>1</Lane2State>
+	<Lane3State>1</Lane3State>
+
+	<!-- Panel Timing -->
+	<PanelTimings>"0x76, 0x18, 0x10, 0x00, 0x3C, 0x40, 0x14, 0x1C, 0x12, 0x03, 0x04, 0x00"</PanelTimings>
+	<DSIMDPTrigger>0</DSIMDPTrigger>
+	<DSIDMATrigger>4</DSIDMATrigger>
+	<TClkPost>0x0a</TClkPost>
+	<TClkPre>0x1d</TClkPre>
+
+	<!-- Backlight -->
+	<BLInterfaceType>1</BLInterfaceType>
+	<BLMinLevel>1</BLMinLevel>
+	<BLMaxLevel>4095</BLMaxLevel>
+	<BLStep>100</BLStep>
+	<BLPMICModel>"PMIC_8941"</BLPMICModel>
+	<BLPMICControlType>1</BLPMICControlType>
+
+	<!-- Panel Reset Sequence -->
+	<ResetSequence>
+		<PinState1>1</PinState1>
+		<PulseWidth1>20</PulseWidth1>
+		<PinState2>0</PinState2>
+		<PulseWidth2>20</PulseWidth2>
+		<PinState3>1</PinState3>
+		<PulseWidth3>20</PulseWidth3>
+		<EnableBit>2</EnableBit>
+	</ResetSequence>
+
+  </PanelEntry>
+</GCDB>
-- 
1.9.1

